### PR TITLE
[microTVM]Add default value to unspecified project options in project API

### PR DESCRIPTION
**Pull request by mehrdadh**
In current TVM, project API ignores the default values that are set for each project option in a project API server. This results in extra function handlers to specify the default value for those options in runtime.
This PR searches through available ProjectOptions for a project API server and adds default values for unspecified options before sending the `options` to API call function handler.

### CODE REVIEW

File: apps/microtvm/arduino/template_project/microtvm_api_server.py [hunk 1]
```diff
@@ -337,7 +332,7 @@ def _copy_cmsis(self, project_path: pathlib.Path, cmsis_path: str):
         However, the latest release does not include header files that are copied in this function.
         """
         (project_path / "include" / "cmsis").mkdir()
-        cmsis_path = get_cmsis_path(cmsis_path)
+        cmsis_path = pathlib.Path(cmsis_path)
```
Comment (areusch) [inline apps/microtvm/arduino/template_project/microtvm_api_server.py:335]: this one also removes the possibility of setting CMSIS_PATH through env right? can you add to PR description?

Comment (mehrdadh) [reply apps/microtvm/arduino/template_project/microtvm_api_server.py:335]: not it doesn't, the default value for cmsis_path is set by environment. Check `python/tvm/micro/project_api/server.py` where we defined the cmsis_path as a projectOption

Comment (areusch) [reply apps/microtvm/arduino/template_project/microtvm_api_server.py:335]: ah, gotcha, nevermind then

**Review by areusch — APPROVED** [review]

**Review by guberti — COMMENTED** [review]
LGTM, sorry for the belated review!

**Pull request merged**